api/turing/api/appcontext.go

```diff
@@ -83,26 +83,27 @@ func NewAppContext(
 		cfg.KubernetesLabelConfigs.Environment,
 	)
 
-	// Initialise Ensembling Job Service
-	ensemblingJobService := service.NewEnsemblingJobService(
-		db,
-		cfg.BatchEnsemblingConfig.JobConfig.DefaultEnvironment,
-		cfg.BatchEnsemblingConfig.ImageBuildingConfig.BuildNamespace,
-		cfg.BatchEnsemblingConfig.LoggingURLFormat,
-		cfg.BatchEnsemblingConfig.MonitoringURLFormat,
-		cfg.BatchEnsemblingConfig.JobConfig.DefaultConfigurations,
-		mlpSvc,
-	)
-
 	// Initialise Batch components
 	// Since there is only the default environment, we will not create multiple batch runners.
 	var batchJobRunners []batchrunner.BatchJobRunner
 
+	var ensemblingJobService service.EnsemblingJobService
 	if cfg.BatchEnsemblingConfig.Enabled {
+		// Initialise Ensembling Job Service
+		ensemblingJobService = service.NewEnsemblingJobService(
+			db,
+			cfg.BatchEnsemblingConfig.JobConfig.DefaultEnvironment,
```

Comment:
The way the validator is written now and with the usage of pointers, `JobConfig` could be `nil` when Enabled=true.

Reply:
Ok I added some checks for JobConfig, RunnerConfig and ImageBuildingConfig